omod/src/main/java/org/openmrs/module/coreapps/page/controller/summarydashboard/ProgramEnrollmentsPageController.java

```diff
@@ -0,0 +1,78 @@
+import org.openmrs.module.appui.UiSessionContext;
+import org.openmrs.module.coreapps.CoreAppsConstants;
+import org.openmrs.module.coreapps.CoreAppsProperties;
+import org.openmrs.module.emrapi.EmrApiProperties;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.openmrs.ui.framework.page.Redirect;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class ProgramEnrollmentsPageController {
+
+    public Object controller(@RequestParam("app") AppDescriptor app,
+                             @SpringBean EmrApiProperties emrApiProperties,
+                             @SpringBean("coreAppsProperties") CoreAppsProperties coreAppsProperties,
+                             @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService,
+                             PageModel model,
+                             UiSessionContext sessionContext) {
+
+        if (!Context.hasPrivilege(CoreAppsConstants.PRIVILEGE_SUMMARY_DASHBOARD)) {
+            return new Redirect("coreapps", "noAccess", "");
+        }
+
+        model.addAttribute("app", app);
+
+        AppContextModel contextModel = sessionContext.generateAppContextModel();
+        model.addAttribute("appContextModel", contextModel);
+
+        List<Extension> includeFragments = appFrameworkService.getExtensionsForCurrentUser(app.getId() + ".includeFragments", contextModel);
+        Collections.sort(includeFragments);
+        model.addAttribute("includeFragments", includeFragments);
+
+        List<Extension> firstColumnFragments = appFrameworkService.getExtensionsForCurrentUser(app.getId() + ".firstColumnFragments", contextModel);
+        Collections.sort(firstColumnFragments);
+        model.addAttribute("firstColumnFragments", firstColumnFragments);
+
+        String programUuid = null;
+        ObjectNode config = app.getConfig();
+        if (config != null) {
+            for (Iterator<Map.Entry<String, JsonNode>> fields = config.getFields(); fields.hasNext();) {
+                Map.Entry<String, JsonNode> field = fields.next();
+                if (StringUtils.equalsIgnoreCase(field.getKey(), "program") ){
+                    programUuid = field.getValue().getTextValue();
+                }
+            }
+        }
+        if (programUuid != null ) {
+            Program program = Context.getProgramWorkflowService().getProgramByUuid(programUuid);
+            if (program != null ) {
+                Date today = Calendar.getInstance().getTime();
+                // retrieve all program enrollments still active as now
+                List<PatientProgram> patientPrograms = Context.getProgramWorkflowService().getPatientPrograms(null, program, null, today, today, null, false);
```

Comment:
I worry about how this will perform for all programs.  I don't really have a better solution in the short term unless we can achieve server-side paging restfully, but just something to watch out for as some programs can have many hundreds or thousands of active patients. I also don't know if we would want to limit this in some way around enrollment location - in our Malawi case, you likely want to see enrollments based on location.